### ISSUE TITLE
Location not published, but paid guest usage possible in combination with price transparency?

### ISSUE THREAD

**Issue by thomashaelsigVAT** [ocpi/ocpi#431]
We (CPO & EMSP in Netherlands) are currently discussing implementation of OCPI 2.2.1 and how to setup out locations where the Owners would not like them visible to the wider public.

We have some employee parking lots/garages that are primarily intended for the employees, but they can use external EMSP cards so we should publish them.

Out thoughts were to set this up like 8.3.1.3. 'Example destination charging location not published, but paid guest usage possible'. 
publish = false
EVSE.parking_restrictions = CUSTOMERS

However there it states that "_Starting a session from an App is not possible, because the driver will not be able to select the Charge Point on a map"._

We were thinking that there is this case where the stations should not appear on the map, but if you search for the EVSE ID, you should be able to find it and see relevant information and even start transaction.
The key point here is around up-front Price Transparency that every Driver using an external EMSP token should be able to see their price, which they can only do by using their EMSP App. This is the situation for the Netherlands but I can assume similar requirements are valid in other countries.

The alternative is to use example 8.3.1.2 and just having a EVSE.parking_restrictions = CUSTOMERS, but we feel that in this case 99.9% of EV drivers can't access the station. Restaurants, stores, supermarkets etc. want their stations to be visible, but in this case the owner would rather spare EV drivers disappointment.

Not publishing at all is also possible, but in the transition phase of OCPI 2.1.1 to 2.2.1 could lead to some issues with CDRs for unknown locations. And it is also not in line with up-front price transparency.

Looking forward to hear your thoughts.
This was the other relevant topic I found around this, but not quite answering the question.
https://github.com/ocpi/ocpi/issues?q=is%3Aissue++publish+

**Comment by reinierl** [comment]
I think there is a general answer and a specific one.

The more general one:

OCPI is a B2B protocol for sharing data with other parties in the EV charging market. Which data you decide to share or not is ultimately a business decision that is not in the scope of OCPI. What exactly the user interface for driver apps will be is not in the scope of OCPI either. A driver app may or may not allow someone to enter an EVSE-ID to gain access to charge points that are not shown on the map. It is fundamentally not OCPI's business either way because that is between the Driver and the MSP or NSP providing the app where the MSP/NSP's product designers set the rules and OCPI does not.

The more specific one:

OCPI 2.2.1 provides many fields in the Location object to signal restrictions on the degree to which the receiving Party should re-share the Location and on who is allowed to charge at the Location.  These fields include the `publish` (a bit confusing, because in a way the CPO has already published the Location as soon as it shares it with other parties, even if it shares with `publish` set to `false`), `publish_allowed_to`, and `evse.parking_restrictions` fields. Your case sounds like the use case for setting `evse.parking_restrictions` to `CUSTOMERS`: charging with roaming cards is possible, but only for people who already have some other relation to the venue and do not visit only to charge there.

I see you already considered a scenario with `publish` = `false`, `EVSE.parking_restrictions` = `CUSTOMERS` but you're afraid that "99.9% of EV drivers can't access the station". I don't fully understand what you mean by that; whether people can access a physical venue is not decided by data transferred over OCPI. That is decided by fences and beams and security staff instead.

I guess you mean "it would probably make it show up in apps, which would draw a large crowd, which is what restaurants and bars want, but not employers, because 99.9% of these drawn crowds would be disappointed because they can't charge there". Then, eh, I'm not sure how apps these days typically deal with `parking_restrictions` set to `CUSTOMERS`. Personally I wouldn't be particularly interested in those if I was looking at a charging app.

Anyway, long story short, @michelbayings do you know if we ever considered `EMPLOYEES` and/or `TENANTS` as possible values for `parking_restrictions` apart from `CUSTOMERS`?

**Comment by mathieu-muller** [comment]
Quick feedback from Chargemap as we are an MSP in France already handling such situations. We have a community backed map (ie users can signal and add charging points). It is part of our job to specify to our users the restrictions for each station. For example, on Tesla super-chargers, we clearly indicates (well that not so true anymore but it was for a long time) that the station was reserved for Tesla car owners. 

Same for some stations that belongs to private users / hotel / restaurants, when the station is not public we clearly indicates the conditions to access the station. And to do so, we also use the data from the parking restrictions in the OCPI protocol.

So, as an MSP, to me it's fine if you publish the location with some parking restrictions. It is our job to tell our customers if they can access one particular location or not :).

Also, it would be great to add more enumerations in the parking restrictions: EMPLOYEES, TENANTS, CUSTOMERS is a good start imho. On our side, we also support:

- RESIDENTS (people that live in the surrounding area)
- MEMBERS (members of a club/organization)
- MANUFACTURER (people with only certain type of cars can charge here, usually used for Tesla chargers)
- AUTHORIZED_PERSONNEL (people will require an additional credential to charge here)
- OTHER

**Comment by thomashaelsigVAT** [comment]
Thanks @reinierl  and @mathieu-muller  for your initial commments.

I think a key issue is that the 'CUSTOMERS' restriction is too broad and can encompass both a relatively accessible 'shoppers only' parking lot at a supermarket but also the very access restricted 'employee only' parking garage.

As CPO we would want to make this difference clear and as an EV Driver I would like to know the difference.

I like both of your thoughts to add additional properties and in the case of ChargeMap already having it implemented through the community. I think that would be a good discussion to have if more Restrictions can be added in the future.

**Comment by reinierl** [comment]
In the OCPI Development Working Group Call at 2023-05-11, we decided that in OCPI 3.0, we will allow:
 * TENANTS and EMPLOYEES as extra possible restrictions
 * specifying if restrictions perhaps apply only during opening hours of the establishment or only outside of those hours
 * time-based restrictions, like "max 2 hours"
